### PR TITLE
add loadflow result to study

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 1]
```diff
@@ -174,8 +174,15 @@ public MockResponse dispatch(RecordedRequest request) {
                     case "/v1/networks/38400000-8cf0-11bd-b23e-10b96e4ef00d/switches/switchId?open=true":
                     case "/v1/networks/38400000-8cf0-11bd-b23e-10b96e4ef00d/run":
                         return new MockResponse().setResponseCode(200)
-                                .addHeader("Content-Type", "application/json; charset=utf-8");
-
+                            .setBody("{\n" +
+                                "\"metrics\":{\n" +
+                                "\"network_0_iterations\":\"7\",\n" +
+                                "\"network_0_status\":\"CONVERGED\"\n" +
```
Comment (niconoir) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:180]: Is it possible to add a test checking that the CONVERGED status value is correctly written in database for the study ?

**Review by niconoir — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 2]
```diff
@@ -322,17 +330,36 @@ public StudyService(
...
             studyUpdatePublisher.onNext(MessageBuilder.withPayload("")
                 .setHeader(STUDY_NAME, studyName)
                 .setHeader(UPDATE_TYPE, UPDATE_TYPE_LOADFLOW)
-                .build())
+                .build()
+            )
         );
     }
 
+    private LoadFlowResult jsonToLoadFlowTesult(String strLfResult) {
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: done

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 1 above)
Comment (JacquesBorsenberger) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:181]: done

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 3]
```diff
@@ -397,12 +424,22 @@ public StudyService(
...
             Mono<Void> deleteStudy = deleteStudy(userId, studyName);
             Mono<Study> insertStudy = insertStudy(newStudyName, userId, study.isPrivate(), study.getNetworkUuid(), study.getNetworkId(),
-                    study.getDescription(), study.getCaseFormat(), study.getCaseUuid(), study.isCasePrivate());
+                    study.getDescription(), study.getCaseFormat(), study.getCaseUuid(), study.isCasePrivate(), study.getLoadFlowResult());
 
             return deleteStudy.then(insertStudy);
         });
     }
 
+    @Transactional
```
Comment (niconoir) [inline src/main/java/org/gridsuite/study/server/StudyService.java]: Transactional annotation is not necessary in this context

Comment (niconoir) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: Already corrected

**Review by niconoir — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 4]
```diff
@@ -374,17 +382,36 @@ public StudyService(
...
                 .setHeader(UPDATE_TYPE, UPDATE_TYPE_LOADFLOW)
-                .build())
+                .build()
+            )
         );
     }
 
+    private LoadFlowResult jsonToLoadFlowResult(String strLfResult) {
+        try {
+            String strStatus = JsonPathUtils.evaluate(strLfResult, "$.metrics.network_0_status");
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/study/server/StudyService.java:402]: @JacquesBorsenberger here you hard code that result is from main connected component, but it should be up to the modele to decide about the status. There is a "ok" attribute for that in loadflow results.

**Review by geofjamg — CHANGES_REQUESTED** [review]
Use LoadFlow.isOk for LF status

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 4 above)
Comment (JacquesBorsenberger) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: done

File: pom.xml [hunk 5]
```diff
@@ -143,6 +143,10 @@
             <artifactId>spring-cloud-stream</artifactId>
         </dependency>
 
+        <dependency>
+          <groupId>com.jayway.jsonpath</groupId>
+          <artifactId>json-path-assert</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply pom.xml:149]: yes

**Review by FranckLecuyer — APPROVED** [review]

**Review by geofjamg — CHANGES_REQUESTED** [review]
It still does not work when I refresh the page during a loadflow run. Button is locked to running state and there is no way to run again.

**Review by geofjamg — APPROVED** [review]

**Pull request merged**